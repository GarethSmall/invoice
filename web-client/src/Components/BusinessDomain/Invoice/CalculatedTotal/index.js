//@flow
import CalculatedTotal from './CalculatedTotal';
export default CalculatedTotal;
