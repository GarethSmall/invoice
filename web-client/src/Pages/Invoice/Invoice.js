// @flow
import React, { PureComponent } from 'react';
import CalculatedTotal from '../../Components/BusinessDomain/Invoice/CalculatedTotal';
import InvoiceTable from '../../Components/BusinessDomain/Invoice/InvoiceTable';
import { Order, Product } from 'shared-flow';

type Props = {};
const mockProducts : Product[] = [
  {
    _id: '1',
    name: 'Test T-shirt',
    price: 32.00,
  },
  {
    _id: '2',
    name: 'Test Jeans',
    price: 75.00,
  }
];
const mockOrder : Order = {
  _id : '1',
  items : [
    {
      _id: '1',
      order: '1',
      product: {
        _id: '1',
        name: 'Test T-shirt',
        price: 32.00,
      },
      quantity: 2,
      price: 64.00,
    },
    {
      _id: '2',
      order: '1',
      product: {
        _id: '2',
        name: 'Test Jeans',
        price: 75.00,
      },
      quantity: 2,
      price: 150.00,
    },
  ],
  subTotalPrice: 214.00,
  taxPrice: 10.70,
  totalPrice: 224.70,
};
export default class Invoice extends PureComponent<Props> {
  render() {
    return [
      <InvoiceTable key="1" products={mockProducts} order={mockOrder} />,
      <CalculatedTotal key="2" items={mockOrder.items} />
    ];
  }
}