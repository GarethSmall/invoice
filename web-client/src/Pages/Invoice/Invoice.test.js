// @flow
import React from 'react';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import shallow from 'enzyme/shallow';
import Invoice from './';
const { describe, test, expect } = global;

Enzyme.configure({ adapter: new Adapter() });

describe('Invoice', () => {
  test('should render', () => {
    const component = shallow((
      <Invoice />
    ));
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
