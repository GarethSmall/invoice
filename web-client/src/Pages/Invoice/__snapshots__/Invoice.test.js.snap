// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice should render 1`] = `
Array [
  <InvoiceTable
    key="1"
    order={
      Object {
        "_id": "1",
        "items": Array [
          Object {
            "_id": "1",
            "order": "1",
            "price": 64,
            "product": Object {
              "_id": "1",
              "name": "Test T-shirt",
              "price": 32,
            },
            "quantity": 2,
          },
          Object {
            "_id": "2",
            "order": "1",
            "price": 150,
            "product": Object {
              "_id": "2",
              "name": "Test Jeans",
              "price": 75,
            },
            "quantity": 2,
          },
        ],
        "subTotalPrice": 214,
        "taxPrice": 10.7,
        "totalPrice": 224.7,
      }
    }
    products={
      Array [
        Object {
          "_id": "1",
          "name": "Test T-shirt",
          "price": 32,
        },
        Object {
          "_id": "2",
          "name": "Test Jeans",
          "price": 75,
        },
      ]
    }
  />,
  <CalculatedTotal
    items={
      Array [
        Object {
          "_id": "1",
          "order": "1",
          "price": 64,
          "product": Object {
            "_id": "1",
            "name": "Test T-shirt",
            "price": 32,
          },
          "quantity": 2,
        },
        Object {
          "_id": "2",
          "order": "1",
          "price": 150,
          "product": Object {
            "_id": "2",
            "name": "Test Jeans",
            "price": 75,
          },
          "quantity": 2,
        },
      ]
    }
    key="2"
    taxPercent={5}
  />,
]
`;
